generation/templates/git_push.sh.mustache

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
```

Comment:
This seems unnecessary.

Reply:
I'm _pretty sure_ (worth testing) you can delete any templates that you did not modify, and it will fallback to the built-in ones. It would be much easier to manage if we have a smaller subset of files we know we need to take care of.
For files that we do modify, it would be useful to commit the original unmodified template, and then commit our changes, so it easy to see our diff. That will make it easier to update templates in the future (if they are changed upstream).

Reply:
I like that.  I'll create a new PR with just the templates that will get modified since we squash individual commits.